src/test/java/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsWorkflowTest.java

```diff
@@ -100,17 +100,20 @@ void basics(JenkinsRule j) throws Throwable {
         WorkflowRun b = p.scheduleBuild2(0).waitForStart();
         SemaphoreStep.waitForStart("waiting/1", b);
         Set<String> expected = new HashSet<>(Arrays.asList("build.xml", "program.dat", "workflow/5.xml"));
+        // Skip assertion on Windows, temporary files contaminate content frequently
         if (!Functions.isWindows()) {
-            // Skip assertion on Windows, temporary files contaminate content frequently
-            assertEquals(expected, grep(b.getRootDir(), "s3cr3t"), "TODO cannot keep it out of the closure block, but at least outside users cannot see this; withCredentials does better");
+            // Compensate for test being flaky on slow filesystems
+            Set<String> actual = grep(b.getRootDir(), "s3cr3t");
+            actual.removeIf(s -> s.matches("^atomic\\d+\\.tmp$"));
```

Comment:
This works, but it seems undesirable to encode knowledge of an internal implementation detail like temporary file creation into the test. Would it work to use Awaitility to wait until the set of results matches the expected set of files? The idea is that we would keep waiting until the temporary file is deleted, at which point we are guaranteed that the operation has finished and the contents of the files are in their final state. To me this is more foolproof than adding a special case for a temporary file, if it works.